app/src/main/java/org/glucosio/android/activity/HelloActivity.java

```diff
@@ -100,11 +104,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
         initStartButton();
 
-        Analytics analytics = application.getAnalytics();
```

Comment:
You can move report analytics to `onStart` of activity and there it will use flag or not. After the dialog is closed it will call this method. This way you can preserve removed test.

Reply:
I'm sorry I don't understand why this is needed

Reply:
We have a bunch of activity lifecycle events in Activity. We are interested in two `onCreate()` and `onStart()`.

`onCreate()` is called only once and there you show dialogue where the user decides to allow or not analytics and crashes.

`onStart()` is called a couple of times and one when the user closes the dialogue. 

So the proposal about moving analytics report to start method will allow to easily test it as well. However, this will bring side effect that this analytics might be reported a couple of times.

Reply:
I will return test and modify it today

Reply:
Addressed